src/server/game/Battlegrounds/Zones/BattlegroundWS.cpp

```diff
@@ -882,7 +882,10 @@ bool BattlegroundWS::CheckAchievementCriteriaMeet(uint32 criteriaId, Player cons
     switch (criteriaId)
     {
         case BG_CRITERIA_CHECK_SAVE_THE_DAY:
-            return GetFlagState(player->GetTeam()) == BG_WS_FLAG_STATE_ON_BASE;
+            if(target && target->GetTypeId() == TYPEID_PLAYER) 
```

Comment:
this line is really necessary? can't think on a situation where this check will fail